packages/orama/src/components/sorter.ts

```diff
@@ -54,7 +61,8 @@ function innerCreate(schema: Schema, sortableDeniedProperties: string[], prefix:
         sorter.sortableProperties.push(path)
         sorter.sortablePropertiesWithTypes[path] = type
         sorter.sorts[path] = {
-          docs: {},
+          docs: new Map(),
```

Comment:
I fear we can't use maps here as they're not directly serializable to JSON. @allevo, @ShogunPanda please correct me if I'm wrong

Reply:
yes correct

```
 % node
Welcome to Node.js v20.2.0.
Type ".help" for more information.
> JSON.stringify({ a: new Map() })
'{"a":{}}'
> a = new Map()
Map(0) {}
> a.set('a', 0)
Map(1) { 'a' => 0 }
> JSON.stringify({ a })
'{"a":{}}'
> a
Map(1) { 'a' => 0 }
> 
```

A custom toJSON function is required here.

Reply:
It's quite strange that our tests couldn't identify this. We need to fix that sooner or later

Reply:
I already cover the serialization here: https://github.com/oramasearch/orama/pull/434/files/698be5fbff8958e80fa30b29465c45deca238cf9#diff-7bf96b26edf177621478ee69fd6da1265f94d1ad031a8468084ec5853ed7e35dR265-R277

Reply:
Correct.